pkg/reconcilermanager/controllers/reconciler_base_test.go

```diff
@@ -488,97 +489,99 @@ spec:
+			// We do not define some containers' resource limits to make the resource burstable.
+			// Config Sync only manages resource limits when a user
+			// defines resource limits through the resource override API.
+			// When a user resets resource limit overrides, the declared deployment will not have
+			// resource limits defined. To make sure a user can always successfully
+			// reset the resource limits override we must track differences in resource limits
+			// between the declared and current deployment, even if no resource limits are defined
+			// in the declared deployment.
+			expectedSame: false,
 		},
 		"non-Autopilot, override limits, no manual change limits": {
 			declared:          declared,
-			unObjCurrent:      unObjCurrent,
+			current:           currentDeploymentUnstructured,
 			isAutopilot:       false,
 			overrideLimit:     true,
 			manualChangeLimit: false,
-			isSame:            false,
+			expectedSame:      false,
 		},
 		"non-Autopilot, override limits, manual change limits": {
 			declared:          declared,
-			unObjCurrent:      unObjCurrent,
+			current:           currentDeploymentUnstructured,
 			isAutopilot:       false,
 			overrideLimit:     true,
 			manualChangeLimit: true,
-			isSame:            false,
+			expectedSame:      false,
 		},
 		"Autopilot, no override limits, no manual change limits": {
 			declared:          declared,
-			unObjCurrent:      unObjCurrent,
+			current:           currentDeploymentUnstructured,
 			isAutopilot:       true,
 			overrideLimit:     false,
 			manualChangeLimit: false,
-			isSame:            true,
+			expectedSame:      true,
 		},
 		"Autopilot, no override limits, manual change limits": {
 			declared:          declared,
-			unObjCurrent:      unObjCurrent,
+			current:           currentDeploymentUnstructured,
 			isAutopilot:       true,
 			overrideLimit:     false,
 			manualChangeLimit: true,
-			isSame:            true,
+			expectedSame:      true,
 		},
 		"Autopilot, override limits, no manual change limits": {
 			declared:          declared,
-			unObjCurrent:      unObjCurrent,
+			current:           currentDeploymentUnstructured,
 			isAutopilot:       true,
 			overrideLimit:     true,
 			manualChangeLimit: false,
-			isSame:            true,
+			expectedSame:      true,
 		},
 		"Autopilot, override limits, manual change limits": {
 			declared:          declared,
-			unObjCurrent:      unObjCurrent,
+			current:           currentDeploymentUnstructured,
 			isAutopilot:       true,
 			overrideLimit:     true,
 			manualChangeLimit: true,
-			isSame:            true,
+			expectedSame:      true,
 		},
 	}
 
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
 			declared := overrideResourceLimits(tc.declared, tc.overrideLimit)
-			unObjCurrent, err := manualChangeResourceLimits(tc.unObjCurrent, tc.manualChangeLimit)
+			currentDeploymentUnstructured, err := manualChangeResourceLimits(tc.current, tc.manualChangeLimit)
```

Comment:
nit: define a new variable inside this loop instead of reusing `currentDeploymentUnstructured` variable that is scoped above. Maybe just
```golang
if tc.overrideLimit {
  tc.declared = overrideResourceLimits(tc.declared)
}
if tc.manualChangeLimit {
  tc.current, err := manualChangeResourceLimits(tc.current)
  require.NoError(t, err)
}
```

Reply:
the idea is to keep the tc.declared(*Deployment) and tc.current(*Unstructured) same for each testcase. So I need to define new variables instead of reuse the tc.declared.